contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -13,20 +13,25 @@ import "./TeleporterRegistry.sol";
  *
  * This contract is intended to be inherited by other contracts that wish to use the
  * upgrade mechanism. It provides a modifier that restricts access to only Teleporter
- * versions that are greater than or equal to `_minTeleporterVersion`.
+ * versions that are greater than or equal to `minTeleporterVersion`.
  */
 abstract contract TeleporterUpgradeable {
     TeleporterRegistry public immutable teleporterRegistry;
-    uint256 internal _minTeleporterVersion;
+    uint256 public minTeleporterVersion;
+
+    event MinTeleporterVersionUpdated(
+        uint256 indexed oldMinTeleporterVersion,
+        uint256 indexed newMinTeleporterVersion
+    );
 
     /**
      * @dev Throws if called by a `msg.sender` that is not an allowed Teleporter version.
-     * Checks that `msg.sender` matches a Teleporter version greater than or equal to `_minTeleporterVersion`.
+     * Checks that `msg.sender` matches a Teleporter version greater than or equal to `minTeleporterVersion`.
      */
-    modifier onlyAllowedTeleporter() {
+    modifier onlyAllowedTeleporter() virtual {
```

Comment:
Is the `virtual` keyword necessary? I'm not sure if modifier overriding is allowed, but I don't think we'd want dapps to do so here.

Reply:
modifiers can be overriden, but you're right that we should remove it for the modifier.